ObjectID Coercion supports array of values

mitsos1os opened the pull request:
### Description

When a property is of type array of 'ObjectId', it now properly is coerced.

#### Related issues

<!--
Please use the following link syntaxes:

- connect to #49 (to reference issues in the current repository)
- connect to strongloop/loopback#49 (to reference issues in another repository)
-->
- fixes #545 

### Checklist

<!--
- Please mark your choice with an "x" (i.e. [x], see
https://github.com/blog/1375-task-lists-in-gfm-issues-pulls-comments)
- PR's without test coverage will be closed.
-->

- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style
  guide](http://loopback.io/doc/en/contrib/style-guide.html)

slnode commented:
Can one of the admins verify this patch? To accept patch and trigger a build add comment ".*ok\W+to\W+test.*"

dhmlau commented:
@mitsos1os, thanks for your PR. Could you please rebase your branch? Thanks!

mitsos1os commented:
Hi @dhmlau , I have rebased and also squashed the commits.
However I came across a bug, that for some reason did not fail the tests earlier.
When using the notation of a type with the extra `mongodb` parameter such as in my tests:
```js
const Article = ds.createModel(
      'ArticleC',
      {
        xid: {type: String, mongodb: {dataType: 'objectid'}},
        xidArr: {type: [String], mongodb: {dataType: 'objectid'}},
        title: String,
      }
    );
```
Although the connector successfully saves the model in the database as an array of ObjectIds, when retrieving the model through `find` then because it is an array of values it is instantiated as a `List` using its declared type *(which is String in this case)*, not taking into account the `mongodb` parameter of the type definition.

This takes place here: https://github.com/strongloop/loopback-datasource-juggler/blob/d54d76947724d75a0f46b151c064742c678b0bbe/lib/model.js#L357

How should we handle this?

mitsos1os commented:
Any news for the PR here?

dhmlau commented:
@mitsos1os, my apologies. I just realized your PR is similar to the PR we just merged: https://github.com/strongloop/loopback-connector-mongodb/pull/560.  What do you think?

mitsos1os commented:
> @mitsos1os, my apologies. I just realized your PR is similar to the PR we just merged: #560. What do you think?

@dhmlau No problem. It fixes the same thing so I will close mine. I know you must have your hands full but it would be great if we could tackle an open PR a bit sooner since there is small number of them open, which makes it quite manageable.

Anyway, thanks for the  info!
